internal/state/execution.go

```diff
@@ -188,6 +188,11 @@ func (blockExec *BlockExecutor) ProcessProposal(
 	}
 
 	resp, err := blockExec.proxyApp.ProcessProposalSync(ctx, req)
+
+	if resp.IsRejected() {
+		blockExec.metrics.ProcessProposalRejected.Add(1)
+	}
```

Comment:
> do you know how much heavier it would be to also record proposer address?

I think it's possible via

```suggestion
	if resp.IsRejected() {
		proposerAddress := block.ProposerAddress.String()
		blockExec.metrics.ProcessProposalRejected.With("proposerAddress", proposerAddress).Add(1)
	}
```

my only concern is that proposerAddress is an unbounded value (in practice < 200) so this metric will increase cardinality. Given the practical limit of proposers is bounded, I'm inclined to add if it will help us. 

Ref warning in: https://prometheus.io/docs/practices/naming/#labels

Reply:
hmmm... maybe we should just log the proposer address instead. That should be sufficient (and something we might already be doing).

we can add this logging in a different PR and issue. Thanks for digging!

Reply:
+1 I had the same idea this morning! I'll remove the proposerAddress from this PR